src/firmware/shared/circular_buffer.c

```diff
@@ -0,0 +1,88 @@
+#include "firmware/shared/circular_buffer.h"
+
+#include <assert.h>
+
+struct CircularBuffer
+{
+    size_t head;
+    size_t tail;
+    size_t max_size;
+    float *buffer_data;
+    bool is_full;
+};
+
+CircularBuffer_t *circular_buffer_create(size_t size)
+{
+    // Check if size is in uint8_t range
+    assert(size > 0 && size < 256);
```

Comment:
We don't need to check max size 
```suggestion
    assert(size > 0);
```

Reply:
Good catch. Removed.